src/parser/jobs/mnk/modules/Star.tsx

```diff
@@ -0,0 +1,91 @@
+import {Plural, Trans} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import Downtime from 'parser/core/modules/Downtime'
+import Suggestions, {SEVERITY, Suggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const RECAST_TIME = 2000
+const SSS_RECAST_TIME = 4000
+
+// TODO:
+// Handle end of fight missed cast?
+// Handle SSS used before LB?
+
+export abstract class Star extends Analyser {
+	static override debug = true
+	static override handle = 'star'
+
+	private starCasts: number[] = []
+
+	@dependency data!: Data
+	@dependency downtime!: Downtime
+	@dependency suggestions!: Suggestions
+
+	override initialise(): void {
+		this.addEventHook(
+			filter<Event>()
+				.source(this.parser.actor.id)
+				.type('action')
+				.action(this.data.actions.SIX_SIDED_STAR.id),
+			this.onCast
+		)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onCast(event: Events['action']): void {
+		this.starCasts.push(event.timestamp)
+	}
+
+	private onComplete(): void {
+		const windows = this.downtime.getDowntimeWindows().filter(window => {
+			this.debug(`${window.end - window.start}`)
+			return (window.end - window.start) > RECAST_TIME
+		})
+
+		let windowsMissed = windows.length
+		let badCasts = 0
+
+		this.starCasts.forEach(timestamp => {
+			const isLastCastBeforeDowntime = this.downtime.isDowntime(timestamp + SSS_RECAST_TIME)
+			if (isLastCastBeforeDowntime) {
+				windowsMissed--
+			} else {
+				badCasts++
+			}
```

Comment:
I don't see you checking the Invulnerability module.
I'd recommend adding a check to see if the player is able to act but the boss is invulnerable, and those should count as good uses.